assets/javascripts/discourse/initializers/chat-setup.js

```diff
@@ -1,28 +1,25 @@
 import { withPluginApi } from "discourse/lib/plugin-api";
 import I18n from "I18n";
 import { bind } from "discourse-common/utils/decorators";
+import { getOwner } from "discourse-common/lib/get-owner";
 
 export default {
   name: "chat-setup",
   initialize(container) {
-    this.currentUser = container.lookup("current-user:main");
     this.chatService = container.lookup("service:chat");
 
     withPluginApi("0.12.1", (api) => {
       // we want to decorate the chat quote dates regardless
       // of whether the current user has chat enabled
       api.decorateCookedElement(
         (elem) => {
-          if (!this.currentUser) {
-            this.currentUser = container.lookup("current-user:main");
-          }
-
-          const currentUserTimezone = this.currentUser?.resolvedTimezone(
-            this.currentUser
-          );
+          const currentUser = getOwner(this).lookup("current-user:main");
```

Comment:
Just curious, isn't  `currentUser` a singleton so looking up here vs during initialisation should yield the same object?

Reply:
It does! But… in every test after that the app instance changes, and the function does not. And it keeps the reference to that first user object / container.

Reply:
Ah well I was close I guess :joy: Thanks @CvX , I only didn't do this lookup here for "efficiency" but guess it wasn't far enough anyway.